runtime/include/QuantumDevice.hpp

```diff
@@ -287,7 +287,7 @@ struct QuantumDevice {
 
      * @return `Result` The measurement result
      */
-    virtual auto Measure(QubitIdType wire) -> Result = 0;
+    virtual auto Measure(QubitIdType wire, int8_t postselect) -> Result = 0;
```

Comment:
This is just a suggestion, but since this is the C++ interface I would also consider making this argument optional. We could either:
- provide the value of `-1` as default (minimal change)
- use an optional type with a null default (a more "modern" approach)

Reply:
Hey folks, just a fly-by, but if we can use a more modern approach as suggested by David for design in future I that would be best across the board. This is fine for now, but using the explicit type-system tags/enums/std::optional/etc for something like this would be a better strategy long term, especially since we have 3 types only we care about and know. I'd propose this is added to the backlog for the API to improve upon in future.

Reply:
We are working on a fix at the moment